sos/report/plugins/instructlab.py

```diff
@@ -0,0 +1,132 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import (Plugin, IndependentPlugin, PluginOpt)
+
+
+class Instructlab(Plugin, IndependentPlugin):
+
+    short_desc = 'Instructlab'
+    plugin_name = 'instructlab'
+    profiles = ('ai',)
+    containers = ('instructlab', 'ilab',)
+    commands = ('ilab',)
+
+    option_list = [
+        PluginOpt('ilab_user', default='', val_type=str,
+                  desc='user that runs instructlab'),
+        PluginOpt('ilab_conf_dir', default='', val_type=str,
+                  desc='instructlab data directory'),
+        PluginOpt('get-cache', default=False,
+                  desc='Capture models and osci cached data')
+    ]
+
+    def setup(self):
+        cont_share_conf_path = '/usr/share/instructlab/config/'
+        cont_opt_path = '/opt/app-root/src'
+        # .cache dir contains the models and oci directories
+        # which can be quite big. We'll gather this only if
+        # specifying it via command line option
+        cache_dir = '/.cache/instructlab'
+        # .config is where the configuration yaml files can
+        # be found. We gather this always.
+        config_dir = '/.config/instructlab'
+        # In the .local directory we can find datasets,
+        # chat logs, taxonomies, and other very useful data
+        # We gather this always.
+        local_share_dir = '/.local/share/instructlab'
+
+        # container paths
+        cont_cache_path = f'{cont_opt_path}{cache_dir}'
+        cont_config_path = f'{cont_opt_path}{config_dir}'
+        cont_local_path = f'{cont_opt_path}{local_share_dir}'
+
+        self.add_forbidden_path([
+            f"{cont_local_path}/taxonomy/.git",
+            f"{cont_local_path}/taxonomy/.github",
+            f"{cont_opt_path}/src/.local/share/instructlab/taxonomy/.git",
+            f"{cont_opt_path}/src/.local/share/instructlab/taxonomy/.github",
+            ])
+
+        subcmds = [
+            'taxonomy diff',
+            'taxonomy diff --taxonomy-base=empty',
+            'system info',
+            'model list',
+            'config show',
+        ]
+
+        data_dirs = [
+            'data',
+            'generated',
+            'taxonomy',
+            'taxonomy_data',
+            'chatlogs',
+            'checkpoints',
+            'datasets',
+            'internal',
+            'phased',
+        ]
+
+        in_container = False
+        container_names = []
+        _containers = self.get_containers()
+
+        for _con in _containers:
+            if _con[1].startswith('instructlab'):
+                in_container = True
+                container_names.append(_con[1])
+
+        if in_container:
```

Comment:
At least in version 1.1 of RHEL AI, instructlab can run both in a container and outside a container, in the same installation. This may change for 1.2 and later